crates/bevy_reflect_compile_fail_tests/tests/reflect_derive/generics.fail.stderr

```diff
@@ -1,16 +1,34 @@
-error[E0599]: no method named `get_field` found for struct `Box<(dyn Reflect + 'static)>` in the current scope
-  --> tests/reflect_derive/generics.fail.rs:16:9
-   |
-16 |     foo.get_field::<NoReflect>("a").unwrap();
-   |         ^^^^^^^^^ method not found in `Box<dyn Reflect>`
-
 error[E0277]: the trait bound `NoReflect: Reflect` is not satisfied
-  --> tests/reflect_derive/generics.fail.rs:14:37
+   --> tests/reflect_derive/generics.fail.rs:16:21
+    |
+16  |     foo.get_field::<NoReflect>("a").unwrap();
+    |         ---------   ^^^^^^^^^ the trait `Reflect` is not implemented for `NoReflect`
+    |         |
+    |         required by a bound introduced by this call
+    |
+    = help: the following other types implement trait `Reflect`:
+              bool
+              char
+              isize
+              i8
+              i16
+              i32
+              i64
+              i128
+            and $N others
+note: required by a bound in `bevy_reflect::GetField::get_field`
+   --> /home/runner/work/bevy/bevy/crates/bevy_reflect/src/struct_trait.rs:242:21
+    |
+242 |     fn get_field<T: Reflect>(&self, name: &str) -> Option<&T>;
+    |                     ^^^^^^^ required by this bound in `GetField::get_field`
+
+error[E0277]: the trait bound `NoReflect: GetTypeRegistration` is not satisfied
+  --> tests/reflect_derive/generics.fail.rs:14:36
    |
-14 |     let mut foo: Box<dyn Reflect> = Box::new(Foo::<NoReflect> { a: NoReflect(42.0) });
-   |                                     ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^ the trait `Reflect` is not implemented for `NoReflect`
+14 |     let mut foo: Box<dyn Struct> = Box::new(Foo::<NoReflect> { a: NoReflect(42.0) });
+   |                                    ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^ the trait `GetTypeRegistration` is not implemented for `NoReflect`
```

Comment:
(non-blocking) annoying that we're lowering the utility of our errors here. i think i ran into this with `TypePath` but i can't remember how i solved it.

Reply:
Yeah I wish we had a way of controlling what errors read so that they could be more useful. Not just for bevy_reflect but lots of other parts of Bevy (*looks at `IntoSystemConfig`*).